psycopg_pool/psycopg_pool/pool_async.py

```diff
@@ -45,14 +45,23 @@ def __init__(
         max_idle: float = 10 * 60.0,
         reconnect_timeout: float = 5 * 60.0,
         reconnect_failed: Optional[
-            Callable[[BasePool[AsyncConnection[Any]]], None]
+            Union[
+                Callable[["AsyncConnectionPool[AsyncConnection[Any]]"], None],
+                Callable[["AsyncConnectionPool[AsyncConnection[Any]]"], Awaitable[None]]
+            ]
         ] = None,
         num_workers: int = 3,
     ):
         self.connection_class = connection_class
         self._configure = configure
         self._reset = reset
 
+        self._reconnect_failed: Callable[
+            ["AsyncConnectionPool[AsyncConnection[Any]]"],
```

Comment:
I think this type should be the `Union` as defined in the parameter. It might be useful to define a `TypeAlias AsyncConnectFailedCB` as alias for the union, and probably the symmetric version `AsyncConnectFailedCB: TypeAlias = Callable[["ConnectionPool[Connection[Any]]"], None]]` on the sync side. I don't think we need to distinguish a `Null` flavour of the callback.

Reply:
Oh, good catch. Do you think there's value in lifting the sync/async type aliases to another file, so they can be shared by the null pools?

Reply:
Perhaps in `base.py`?

Reply:
Ah, no, too many wrong way imports in `base.py` to make that work.

Reply:
Also fixed now.

Reply:
Oh, I need to fix the sync type reuse as well, whoops

Reply:
Okay, now it's done